tests/wrappers/test_sidebias.py

```diff
@@ -0,0 +1,186 @@
+import re
+import sys
+from unittest.mock import Mock
+
+import numpy as np
+import pytest
+
+# Add the neurogym path if needed
+sys.path.insert(0, "/Users/giuliacrocioni/Desktop/docs/eScience/projects/ANNUBeS/neurogym")
+
+import neurogym as ngym
+from neurogym.wrappers.side_bias import SideBias
+
+
+class MockTrialEnv(ngym.TrialEnv):
+    """Mock environment for testing SideBias wrapper."""
+
+    def __init__(self):
+        super().__init__()
+        self.choices = [0, 1]  # Binary choice task
+        self.observation_space = Mock()
+        self.action_space = Mock()
+
+    def _new_trial(self, **kwargs):
+        return {"ground_truth": kwargs.get("ground_truth", 0)}
+
+    def _step(self, _action):
+        return np.array([0, 0]), 0, False, False, {"new_trial": True}
+
+
+def test_probs_validation():
+    """Test that probs parameter validation works correctly."""
+    env = MockTrialEnv()
+
+    msg = "probs must be a 2D list of lists (matrix) with shape (n_blocks, n_choices)"
+
+    # Test 1: None probs should raise ValueError
+    with pytest.raises(TypeError, match=re.escape(msg)):
+        SideBias(env, probs=None)
+
+    # Test 2: Float/int probs should raise TypeError (no auto-generation)
+    with pytest.raises(TypeError, match=re.escape(msg)):
+        SideBias(env, probs=0.8)
+
+    # Test 3: Wrong dimensions should raise ValueError
+    with pytest.raises(TypeError, match=re.escape(msg)):
+        SideBias(env, probs=[0.8, 0.2])  # 1D array
+
+    # Test 4: Wrong number of choices should raise ValueError
+    with pytest.raises(ValueError):
+        SideBias(env, probs=[[0.8, 0.1, 0.1], [0.2, 0.4, 0.4]])  # 3 choices, env has 2
+
+    # Test 5: Rows not summing to 1 should raise ValueError
+    with pytest.raises(ValueError):
+        SideBias(env, probs=[[0.8, 0.1], [0.2, 0.9]])  # First row sums to 0.9, second to 1.1
+
+    env.choices = [0, 1, 2]
+    # Test 6: Negative probabilities should raise ValueError
+    with pytest.raises(ValueError, match="All probabilities in probs must be non-negative."):
```

Comment:
I generally don't like adding specific error messages, because these might change in the future. Also, you did not do this for any of the other checks.